src/steamship/agents/schema/cache.py

```diff
@@ -0,0 +1,187 @@
+import hashlib
+import json
+import logging
+from typing import Dict, List, Optional
+
+from steamship import Block, MimeTypes, Steamship
+from steamship.agents.schema.action import Action, FinishAction
+from steamship.utils.kv_store import KeyValueStore
+
+
+def _blocks_to_cache_dict(value: List[Block]) -> Dict[str, any]:
+    """Attempts to convert a list of blocks to key-store-safe dictionary.
+
+    Convention: {'blocks':[{'id': <block_id>}, {'text': <some_text>}]}
+    """
+
+    if not value:
+        return {}
+    if not isinstance(value, list):
+        return {}
+    blocks = []
+    for block in value:
+        if block.id:
+            blocks.append({"id": block.id})
+        else:
+            # TODO(dougreid): safe assumption about temporary blocks?
+            blocks.append({"text": block.text})
+    return {"blocks": blocks}
+
+
+def _blocks_from_cache_dict(client: Steamship, value: Dict[str, any]) -> List[Block]:
```

Comment:
Really curious why serializing the whole block didn't work out. You mentioned the `urls` not serializing? 

You're probably way ahead of me on this, but could we just add that into the JSON structure?

Reply:
maybe. i think they are just python properties at this point. i got a little lost in the expectations of serialization and rather than try to fix that, i just tried to work around it.

worth addressing again, especially if it causes latency issues.